plugin/entrySchema.go

```diff
@@ -191,17 +221,85 @@ func (s *EntrySchema) fillChildren(visited map[string]bool) {
 		// true if s is e.g. a volume directory.
 		return
 	}
-	children := s.entry.(Parent).ChildSchemas()
+
+	// Fill-in the meta attribute + metadata schemas
+	var err error
+	if s.metaAttributeSchemaObj != nil {
+		s.MetaAttributeSchema, err = s.schemaOf(s.metaAttributeSchemaObj)
+		if err != nil {
+			s.fillPanicf("bad value passed into SetMetaAttributeSchema: %v", err)
+		}
+	}
+	if s.metadataSchemaObj != nil {
+		s.MetadataSchema, err = s.schemaOf(s.metadataSchemaObj)
+		if err != nil {
+			s.fillPanicf("bad value passed into SetMetadataSchema: %v", err)
+		}
+	}
+
 	visited[s.TypeID()] = true
+
+	// "sParent" is read as "s.parent"
+	sParent := s.entry.(Parent)
+	children := sParent.ChildSchemas()
 	for _, child := range children {
 		if child == nil {
-			msg := fmt.Sprintf("s.fillChildren: found a nil child schema for %v", s.TypeID())
-			panic(msg)
+			s.fillPanicf("found a nil child schema")
 		}
 		s.entrySchema.Children = append(s.entrySchema.Children, child)
-		child.fillChildren(visited)
+		passAlongWrappedTypes(sParent, child.entry)
+		child.fill(visited)
 	}
 	// Delete "s" from visited so that siblings or ancestors that
 	// also use "s" won't be affected.
 	delete(visited, s.TypeID())
 }
+
+// This helper's used by CachedList + EntrySchema#fill(). The reason for
+// the helper is because /fs/schema uses repeated calls to CachedList when
+// fetching the entry, so we need to pass-along the wrapped types when
+// searching for it. However, Parent#ChildSchemas uses empty Entry objects
+// that do not go through CachedList (by definition). Thus, the entry found
+// by /fs/schema needs to pass its wrapped types along to the children to
+// determine their metadata schemas. This is done in s.fill().
+func passAlongWrappedTypes(p Parent, child Entry) {
+	var wrappedTypes SchemaMap
+	// This check is equivalent to "_, ok := child.(Root); ok".
+	// However the latter's more expensive due to the extra
```

Comment:
Have you benchmarked whether this is true?

Reply:
Looking at the assembly produced, it seems like the same amount of work.

Reply:
Also seems simpler if `Root` just overrode the implementation of `wrappedTypes` to call `WrappedTypes` (if it exists).

Reply:
Nevermind, forgot Root was an interface. What's here now looks good.